vivarium/interface/interactive.py

```diff
@@ -0,0 +1,120 @@
+
+    def run_until(self, end_time, with_logging=True):
+        if not isinstance(end_time, type(self.clock.time)):
+            raise ValueError(f"Provided time must be an instance of {type(self.clock.time)}")
+        iterations = int(ceil((end_time - self.clock.time)/self.clock.step_size))
+        self.take_steps(number_of_steps=iterations, with_logging=with_logging)
+        assert self.clock.time - self.clock.step_size < end_time <= self.clock.time
+        return iterations
+
+    def step(self, step_size=None):
+        old_step_size = self.clock.step_size
+        if step_size is not None:
+            if not isinstance(step_size, type(self.clock.step_size)):
+                raise ValueError(f"Provided time must be an instance of {type(self.clock.step_size)}")
+            self.clock._step_size = step_size
+        super().step()
+        self.clock._step_size = old_step_size
+
+    def take_steps(self, number_of_steps=1, step_size=None, with_logging=True):
+        if not isinstance(number_of_steps, int):
+            raise ValueError('Number of steps must be an integer.')
+
+        if run_from_ipython() and with_logging:
+            for _ in log_progress(range(number_of_steps), name='Step'):
+                self.step(step_size)
+        else:
+            for _ in range(number_of_steps):
+                self.step(step_size)
+
+    def list_values(self):
+        return list(self.values.keys())
+
+    def get_values(self):
+        return self.values.items()
+
+    def get_value(self, value_pipeline_name):
+        return self.values.get_value(value_pipeline_name)
+
+    def list_events(self):
+        return self.events.list_events()
+
+    def get_listeners(self, event_name):
+        if event_name not in self.events:
+            raise ValueError(f'No event {event_name} in system.')
+        return self.events.get_listeners(event_name)
+
+    def get_emitter(self, event_name):
+        if event_name not in self.events:
+            raise ValueError(f'No event {event_name} in system.')
+        return self.events.get_emitter(event_name)
+
+    def get_components(self):
+        return [component for component in
+                self.component_manager._components + self.component_manager._managers + self.component_manager._globals]
+
+    def reload_component(self, component):
+        raise NotImplementedError()
+
+    def replace_component(self, old_component, new_component):
+        self.component_manager._components.remove(old_component)
+        new_component.setup(self.builder)
+        self.component_manager.add_components([new_component])
+
+
+def setup_simulation(components, input_config=None):
+    config = build_simulation_configuration()
+    config.update(input_config)
+
+    simulation = InteractiveContext(config, components)
+    simulation.setup()
+    simulation.initialize_simulants()
+
+    return simulation
+
+
+def setup_simulation_from_model_specification(model_specification_file):
```

Comment:
It might be nice to allow this to take a yaml string as well as a path. I suspect that the yaml string is the more useful option, but maybe not.

Reply:
Do you have a specific use case in mind?  This is the feature as abie requested it.